presto-spi/src/main/java/io/prestosql/spi/block/LazyBlock.java

```diff
@@ -275,6 +281,15 @@ public Block getLoadedBlock()
         return lazyData.getFullyLoadedBlock();
     }
 
+    @Override
+    public Block loadSelectedPositions(SelectedPositions selectedPositions)
```

Comment:
Can we convert get positions and get region to this call, and drop some of the code?